packages/extension-vscode/src/server.ts

```diff
@@ -207,6 +207,48 @@ connection.onInitialize((params) => {
     // TODO: Support multiple workspaces (`params.workspaceFolders`).
     workspace = params.rootPath || '';
 
+    /**
+     * The `local` connector uses `jsdom` that is going to always try to load
+     * `canvas` even though it is not needed in the VS Code extension. If there
+     * is a mismatch between the user's node version and the one uses by
+     * VS Code, `canvas` will fail to load and crash the extension. To avoid
+     * that we hijack `require`'s cache and set an empty `Module` so `jsdom`
+     * doesn't use it and continues executing normally.
+     *
+     * Because `jsdom` is the one requireing `canvas` we search for `canvas`
+     * as if we were that package to make sure to find the same package.
+     *
+     * The path nam resolution code is inspired by `Module._resolveFilename`
+     * https://github.com/nodejs/node/blob/5c2d555b29d99f9d1f484fd46eff33b42ee9c11f/lib/internal/modules/cjs/loader.js#L569
+     */
+    const Module = require('module');
+    const fakeParent = new Module('', null);
+
+    fakeParent.paths = Module._nodeModulePaths(workspace);
+
```

Comment:
👆 If you can do this with `require.resolve(request, options)` and the _(`path` options)_ that would be 👌

Reply:
Working 🎉